### PR TITLE
Add gauge output documentation for 5.4.0

### PR DESCRIPTION
**Pull request by mandli**
Add gauge output option documentation.

### CODE REVIEW

File: doc/gauges.rst [hunk 1]
```diff
@@ -66,9 +64,43 @@ where level is the AMR level used to determine the q values at this time.
 Internally the finest level available at each gauge is used, with bilinear
 interpolation to the gauge locations from the 4 nearest cell centers.
 
-If you wish to change what is output at these points, you should copy the library
-routine `dumpgauge.f` to your own directory and modify it appropriately.
+.. If you wish to change what is output at these points, you should copy the library
+.. routine `dumpgauge.f` to your own directory and modify it appropriately.
```
Comment (rjleveque) [inline doc/gauges.rst]: There is no longer a file `dumpgauge.f`, it was replaced by `gauges_module.f90` back in 5.3.0.

File: doc/gauges.rst [hunk 2]
```diff
@@ -66,9 +64,43 @@ where level is the AMR level used to determine the q values at this time.
...
+
+The output that is in the gauge files can be controlled by a variety of parameters.  These can be specified on a per gauge basis or set for all gauges specified.  The output parameters are
+
+- *file_format* : Specifies the file format of the gauge data.  Currently
+  *"ASCII"* is the only value accepted.
```
Comment (rjleveque) [inline doc/gauges.rst]: In setrun and setplot we use `format="ascii"` (lower case).  Is this also allowed for gauges?

File: doc/gauges.rst [hunk 3]
```diff
@@ -66,9 +64,43 @@ where level is the AMR level used to determine the q values at this time.
...
+- *display_format* : Specifies the format of the numbers written to the gauge
+  file for each field.  These are Fortran format strings defaulting to
+  *"e15.7"*.
+- *q_out_fields* : Specifies which fields of the q array to output. Specify as
```
Comment (rjleveque) [inline doc/gauges.rst]: I assume this is Python 0-based indexing?  We should clarify.

File: doc/gauges.rst [hunk 4]
```diff
@@ -66,9 +64,43 @@ where level is the AMR level used to determine the q values at this time.
...
+  a list the indices that should be output.  Defaults to *"all"*.
+- *aux_out_fields* : Specifies which fields of the aux array to output.
+  Specify as a list the indices that should be output. Defaults to *"none"*
+- *min_time_increment* : Specify a minimum amount of time that should pass
+  before recording the values at a gauge.  This can be useful for decreasing
+  the amount of output at a gauge location that is currently being time-
```
Comment (rjleveque) [inline doc/gauges.rst]: Line shouldn't split at hyphen.

File: doc/gauges.rst [hunk 5]
```diff
@@ -66,9 +64,43 @@ where level is the AMR level used to determine the q values at this time.
...
+*aux_out_fields* one can also specify *"all"* to output all fields or *"none"*
+to specify none of them (equivalent to an empty list of indices).
 
+**Note** for GeoClaw that we also output the sea-surface value after the q
+fields.
+In the case of the multilayer code the eta for each surface follows the q
+fields for that layer.
 
 .. warning:: When doing a restart, previous gauge output is deleted unless
    you are careful to preserve it.  See :ref:`restart_output`.
```
Comment (rjleveque) [inline doc/gauges.rst]: The file `restart_output.rst` is also out of date -- it still refers to the file `fort.gauge`.   Presumably the concatenation recommended there now has to be done for each `gauge.000x` file.

Have we checked that gauge restarts still work properly modulo this?  Is there a better way to handle restarts?

File: doc/release_5_4_0.rst [hunk 6]
```diff
@@ -41,8 +41,18 @@ required only one master list needs updating rather than the
 See :ref:`makefiles_library` for more details on how to specify
 local files in place of default library files.
 
-**Better support for gauges.**
-@mandli will write something.
+**Improved Gauge Output Options**
+Gauges now support a number of additional output options including:
+ - specification of output fields, i.e. you can now specify the q and aux
```
Comment (rjleveque) [inline doc/release_5_4_0.rst]: Need a blank line?  I get an unexpected indentation error from Sphinx.

File: doc/release_5_4_0.rst [hunk 7]
```diff
@@ -41,8 +41,18 @@ required only one master list needs updating rather than the
...
+   fields that are output;
+ - specification of output field format, i.e. you can now specify the number
+   of digits to output;
+ - a minimum length of time at which a gauge is allowed to output, i.e. if
+   this was set to 10 then the gauge would only output every 10 seconds or
```
Comment (rjleveque) [inline doc/release_5_4_0.rst]: In geoclaw time is in seconds but perhaps not in other applications.  Say "every 10 time units" here?

File: doc/release_5_4_0.rst [hunk 8]
```diff
@@ -41,8 +41,18 @@ required only one master list needs updating rather than the
...
+   longer;
+ - support for future file format specifications (only ASCII is supported now);
+ - some header info to reflect what is in the file has been added; and
+ - a refactor of how the code stores gauge data (there is now a new `gauge_type`).
```
Comment (rjleveque) [inline doc/release_5_4_0.rst]: In what module?  Users might think you mean `gauge_type` is something to specify in `setrun.py` and not find it in the docs.

**Review by rjleveque — COMMENTED** [review]
Thanks @mandli!

File: doc/gauges.rst (hunk 1 above)
Comment (mandli) [inline doc/gauges.rst]: I just commented out those lines but I suppose we should just remove them.

File: doc/gauges.rst (hunk 2 above)
Comment (mandli) [inline doc/gauges.rst]: Yeah, it's case insensitive.  I can change this.

File: doc/gauges.rst (hunk 5 above)
Comment (mandli) [inline doc/gauges.rst]: No idea.  In principle nothing has changed from @mjberger's initial changes so if they were changed there we should be ok.  Of course restarts would be incompatible between 5.4 and previous versions.

**Pull request merged**